core/src/main/java/com/scalar/db/storage/cosmos/CosmosOperationChecker.java

```diff
@@ -2,36 +2,118 @@
 
 import com.scalar.db.api.ConditionalExpression;
 import com.scalar.db.api.Delete;
+import com.scalar.db.api.Get;
 import com.scalar.db.api.Mutation;
+import com.scalar.db.api.Operation;
 import com.scalar.db.api.Put;
+import com.scalar.db.api.Scan;
 import com.scalar.db.api.TableMetadata;
 import com.scalar.db.common.TableMetadataManager;
 import com.scalar.db.common.checker.OperationChecker;
 import com.scalar.db.common.error.CoreError;
 import com.scalar.db.config.DatabaseConfig;
 import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.io.BigIntColumn;
+import com.scalar.db.io.BlobColumn;
+import com.scalar.db.io.BooleanColumn;
+import com.scalar.db.io.ColumnVisitor;
 import com.scalar.db.io.DataType;
+import com.scalar.db.io.DoubleColumn;
+import com.scalar.db.io.FloatColumn;
+import com.scalar.db.io.IntColumn;
+import com.scalar.db.io.TextColumn;
 
 public class CosmosOperationChecker extends OperationChecker {
+
+  private static final ColumnVisitor PRIMARY_KEY_COLUMN_CHECKER =
+      new ColumnVisitor() {
+        @Override
+        public void visit(BooleanColumn column) {}
+
+        @Override
+        public void visit(IntColumn column) {}
+
+        @Override
+        public void visit(BigIntColumn column) {}
+
+        @Override
+        public void visit(FloatColumn column) {}
+
+        @Override
+        public void visit(DoubleColumn column) {}
+
+        @Override
+        public void visit(TextColumn column) {
+          String value = column.getTextValue();
+          assert value != null;
+
+          if (value.indexOf(':') != -1
```

Comment:
Should the invalid characters be managed as a list and defined as a static variable?
(here, we can do a simple loop to check if a given column includes the characters.)

The list can be joined and also used by the message for `COSMOS_PRIMARY_KEY_CONTAINS_ILLEGAL_CHARACTER` so that we don't have to maintain the characters in different places separately?

Reply:
@feeblefakie Thank you. Fixed in 4a226599cb071027749a1376a6469a5a3e92d4d9.

Regarding the message for `COSMOS_PRIMARY_KEY_CONTAINS_ILLEGAL_CHARACTER`, I didn’t make any modifications in this commit. If we make the illegal characters a variable, the message in the error code document appears as follows:
```
The value of the column %s in the primary key contains an illegal character. Primary key columns must not contain any of the following characters: %s. Value: %s
```

As you can see, it doesn’t show which characters are illegal; it only displays `%s`. That’s why I chose not to replace it with a variable. What do you think?

Reply:
I might miss your point, but I left a suggestion. PTAL.
https://github.com/scalar-labs/scalardb/pull/2292#discussion_r1820525473